prbt_hardware_support/include/prbt_hardware_support/operation_mode_filter.h

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2019 Pilz GmbH & Co. KG
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPERATION_MODE_FILTER_H
+#define OPERATION_MODE_FILTER_H
+
+#include <vector>
+
+#include <message_filters/simple_filter.h>
+
+#include <prbt_hardware_support/modbus_msg_operation_mode_wrapper.h>
+#include <prbt_hardware_support/modbus_msg_operation_mode_wrapper_exception.h>
+#include <prbt_hardware_support/modbus_api_spec.h>
+#include <prbt_hardware_support/opertion_modes.h>
+
+namespace message_filters
+{
+
+/**
+ * @brief Filter to be applied to modbus messages. Only the messages with changes in the brake test state are passed.
+ *        The filter also checks whether the message can be wrapped.
+ *
+ * \code
+ *   message_filters::Subscriber sub<MsgStamped>(nh, TOPIC_NAME, 1);
+ *   message_filters::OperationMode filter<MsgStamped>(&sub));
+ *
+ *   // Register all callback receiving filtered output
+ *   filter.registerCallback(&filteredCallback);
+ * \endcode
+ */
+template <typename M>
```

Comment:
Why do we need this template? We are dealing with Modbus-Msgs and we know this because in the `cb()` function it is even hard coded by the fact that the `ModbusMsgOperationModeWrapper` is used. Therefore, is suggest to remove the template here and derive the class from `SimpleFilter<ModbusMsgInStamped>` instead of `SimpleFilter<M>`.